src/hdr_histogram.c

```diff
@@ -367,10 +386,12 @@ int hdr_alloc(int64_t highest_trackable_value, int significant_figures, struct h
 /* reset a histogram to zero. */
 void hdr_reset(struct hdr_histogram *h)
 {
+    hdr_mutex_lock(&hdrMutex);
```

Comment:
Note: Given that `total_count`, `min_value` and `max_value` are all atomic (and moreover you access them without a mutex elsewhere), you might as well constrain the mutex scope to just around the `memset()` of `h->counts`.

Reply:
Furthermore, given h->counts can again be modified without a lock by other threads , you could just remove the mutex entirely and simply clear counts using a for() loop using atomic_store_explicit() on each element.

Reply:
Done